source/setup/nao_setup_ubuntu.rst

```diff
@@ -0,0 +1,226 @@
+
+You can also use the flash script in the bin directory to flash the USB.  A help menu will be displayed without args.
+    .. code-block:: bash
+
+        ./bin/make-usb.sh
+
+
+Once the USB is made, turn off the robot and plug the USB to the back of its head. Then, hold the chest button continuously until it turns blue.
+
+Let go, and it should start flashing blue rapidly. The lights on the side of the head will tell you the flashing progress. Once it's done, the robot will automatically boot.
+
+.. tip::
+    If this step takes too long (>30 minutes) turn the robot off and retry flashing.
+
+Network flash
+****************
+You can also flash over the network with the following script:
+
+    .. code-block:: bash
+
+        ./bin/flash-robot.sh <robot-hostname or ip>
+
+Restarting the robot will then begin the flashing process (the script should also prompt you to do so)
+
+
+*********************************
+Robot Config, Name and Wifi Setup
+*********************************
+**If the robots are new** you will need to:
+
+* Add the robot to ``robots/robots.cfg``
+    * You can find its head id via ``cat /sys/qi/head_id``.
+    * If you don't know it yet, you can flash without adding the robot to this file, and discover the head ID by sshing to the IP it calls out after flashing.
+
+* Add ``<robot-name>`` to the list of robots in ``utils/webnao/src/common/dicts/robots.ts``
+* Create a copy of the default ``.cfg`` file called ``<robot-name>.cfg`` in ``image/home/nao/data/configs/``
+* Create a copy of the default ``.cfg`` file called ``<robot-name>.cfg`` in ``image/home/nao/data/configs/body/``
+
+*********************************
+Uploading rUNSWift
+*********************************
+
+The image by default should already have rUNSWift. You can check details about the flash by looking at ``image.commit.sha`` and ``image.build.time`` in the home dir.
+
+To flash a new version, run:
+
+.. code-block:: bash
+
+    bin/nao_sync.sh <robot hostname or ip>
+
+The robot should already be good to go with the IP specified in robots/robots.cfg for LAN and wifi in SPL_A
+
+You can see the network settings in ``/etc/netplan`` directory. You can modify these files and then run ``sudo netplan apply`` to apply the changes.
+
+However, we do have scripts to manage these files as we detail further down below. Therefore, you shouldn't really need to modify them yourself.
+
+The default username and password for the robot after flashing is ``nao:nao``.
+Save yourself the hassle of typing this out repeatedly and add your key to ``image/home/nao/.ssh/authorized_keys`` and flash.
+
+*********************************
+Troubleshooting
+*********************************
+
+.. tip::
+
+    It's sometimes useful to just flash the base opn image if you experience issues with robot kinematics, but are confident there are no hardware issues.
+    You can simply use the base image with the existing flash methods to restore the default nao image.
```

Comment:
"and then attempt to flash the custom image once you confirm things roughly work as expected in the base image"